src/vanflow.c

```diff
@@ -1829,6 +1851,9 @@ vflow_record_t *vflow_start_record_custom_id(vflow_record_type_t record_type, vf
 //=====================================================================================
 vflow_record_t *vflow_start_record(vflow_record_type_t record_type, vflow_record_t *parent)
 {
+    if (observability_records[record_type] && state->emit_observability_records == false) {
```

Comment:
Is `==` a valid operator for atomics?  Does this work?  I think you need to use `sys_atomic_get` to retrieve the value of the counter.